src/c++/perf_analyzer/test_profile_data_exporter.cc

```diff
@@ -0,0 +1,181 @@
+            "experiment" : {
+              "mode" : "concurrency",
+              "value" : 4
+            },
+            "requests" : [
+              {
+                "timestamp" : 1,
+                "sequence_id" : 1,
+                "response_timestamps" : [ 2, 3 ]
+              }
+            ],
+            "window_boundaries" : [ 1, 5, 6 ]
+          }
+        ],
+        "version" : "1.2.3"
+      }
+      )"};
+
+    rapidjson::Document expected_document;
+    expected_document.Parse(json.c_str());
+
+    const rapidjson::Value& expected_experiment{
+        expected_document["experiments"][0]["experiment"]};
+    const rapidjson::Value& expected_request{
+        expected_document["experiments"][0]["requests"][0]};
+    const rapidjson::Value& expected_windows{
+        expected_document["experiments"][0]["window_boundaries"]};
+    const rapidjson::Value& expected_version{expected_document["version"]};
+
+    const rapidjson::Value& actual_experiment{
+        exporter.document_["experiments"][0]["experiment"]};
+    const rapidjson::Value& actual_request{
+        exporter.document_["experiments"][0]["requests"][0]};
+    const rapidjson::Value& actual_windows{
+        exporter.document_["experiments"][0]["window_boundaries"]};
+    const rapidjson::Value& actual_version{exporter.document_["version"]};
+
+    CHECK(actual_experiment["mode"] == expected_experiment["mode"]);
+    CHECK(actual_experiment["value"] == expected_experiment["value"]);
+
+    CHECK(actual_request["timestamp"] == expected_request["timestamp"]);
+    CHECK(actual_request["sequence_id"] == expected_request["sequence_id"]);
+    CHECK(
+        actual_request["response_timestamps"][0] ==
+        expected_request["response_timestamps"][0]);
+    CHECK(
+        actual_request["response_timestamps"][1] ==
+        expected_request["response_timestamps"][1]);
+
+    CHECK(actual_windows[0] == expected_windows[0]);
+    CHECK(actual_windows[1] == expected_windows[1]);
+    CHECK(actual_windows[2] == expected_windows[2]);
+
+    CHECK(actual_version == expected_version);
+  }
+
+  SUBCASE("AddExperiment")
+  {
+    Experiment raw_experiment;
+    rapidjson::Value entry(rapidjson::kObjectType);
+    rapidjson::Value experiment(rapidjson::kObjectType);
+
+    SUBCASE("Concurrency mode")
+    {
+      InferenceLoadMode infer_mode{15, 0.0};
+      raw_experiment.mode = infer_mode;
+
+      exporter.AddExperiment(entry, experiment, raw_experiment);
+      CHECK(entry.HasMember("experiment"));
+      CHECK(entry["experiment"]["mode"] == "concurrency");
+      CHECK(entry["experiment"]["value"] == 15);
+    }
+
+    SUBCASE("Request rate mode")
+    {
+      InferenceLoadMode infer_mode{0, 23.0};
+      raw_experiment.mode = infer_mode;
+
+      exporter.AddExperiment(entry, experiment, raw_experiment);
+      CHECK(entry.HasMember("experiment"));
+      CHECK(entry["experiment"]["mode"] == "request_rate");
+      CHECK(entry["experiment"]["value"] == 23);
```

Comment:
Make 23.5 and fix `ProfileDataExporter::AddExperiment()` to use SetDouble for request rate case